src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -53,8 +56,68 @@ namespace System.Net.Http {
 #else
 namespace Foundation {
 #endif
+
+	// useful extensions for the class in order to set it in a header
+	static class NSHttpCookieExtensions
+	{
+		private static bool AppendSegment(StringBuilder builder, bool first, string name, string value)
```

Comment:
Maybe `first` should be passed by ref, so we ensure we don't forget to assign it to first when calling it? Makes all the calls simpler too.

Reply:
All the `first` logic can be replaced by:

```csharp
if (builder.Length == 0)
    builder.Append ("; ");
```